### DISCUSSION TITLE
Silence the protoc plugin

### DISCUSSION THREAD

**Discussion by jan-xyz** [square/wire#2091] [Q&A]
Thanks for this awesome plugin!

Is there a way to disable the log messages `Writing com.example.foo to ./src/main/kotlin` when using the wire protoc plugin?

**Comment by oldergod** [comment]
This will be fixed in 4.0 where logs such as the one you quote will be raised to information level (they are right now debug level I believe)

**Reply by jan-xyz** [reply]
I just tried it with the new 4.0.0 version and the same logging persists. Is there a way to control the log level when using it as a protoc plugin?

e.g. this is how I currently execute it:
```
        java -jar /path/to/wire/protoc-plugin \
            --proto_path=protos/schemas \
            --proto_path=./ \
            --kotlin_out=wire-kotlin/$packageName/src/main/kotlin \
            $(find protos/schemas/${directory} -type f -name '*.proto' | sed -e "s:^protos/schemas/::")
```

**Reply by oldergod** [reply]
I didn't realise you were calling Wire via CLI. If you're using Wire this way, there are no way to remove those. Could you not use Gradle to execute Wire?
Anyhow I intend on refactoring how Wire does logging in 4.1. We might allow the CLI option to pass a logger class or something to change that.

**Reply by jan-xyz** [reply]
It's currently part of a larger internal pipeline to build artefacts for all our APIs for different languages. I am also not a Gradle expert, but is there an equivalent to what I'm doing above? Maybe there is a way that we can already refactor this part.

**Reply by oldergod** [reply]
The code you're using is simply calling that method https://github.com/square/wire/blob/master/wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt#L239

You could create another entry point by directly calling `WireCompiler` or `WireRun`.
